File: monorepo_csharp/apps/Web/CycleA.cs
namespace Mono.Web;

public class CycleA
{
    private readonly string _name;
    public CycleB? Child { get; set; }                        // C1.a.type_ref → CycleB

    public CycleA(string name) { _name = name; }

    public CycleB SpawnBravo() => new CycleB(_name + "/b");
    public string Describe()   => $"CycleA({_name})";

    public static string KickOff()
    {
        var a = new CycleA("root");
        var b = a.SpawnBravo();
        return b.BounceToAlpha();
    }
}

public class CycleB
{
    private readonly string _tag;
    public CycleA? Owner { get; set; }                        // C1.b.type_ref → CycleA

    public CycleB(string tag) { _tag = tag; }

    public string BounceToAlpha() => new CycleA($"bounce-from-{_tag}").Describe();
}

File: monorepo_php/apps/web/src/CycleA.php
<?php
namespace Mono\Web;

class CycleA {
    public ?CycleB $child = null;                    // C1.a.type_ref → CycleB

    public function __construct(public readonly string $name) {}
    public function describe(): string { return "CycleA({$this->name})"; }
    public function spawnBravo(): CycleB { return new CycleB("{$this->name}/b"); }

    public static function kickOff(): string {
        $a = new self("root");
        $b = $a->spawnBravo();
        return $b->bounceToAlpha();
    }
}
